src/Microsoft.Azure.WebJobs.Extensions.Kafka/Listeners/KafkaListener.cs

```diff
@@ -65,14 +73,13 @@ internal class KafkaListener<TKey, TValue> : IListener
             this.requiresKey = requiresKey;
             this.logger = logger;
             this.cancellationTokenSource = new CancellationTokenSource();
+            this.consumerGroup = string.IsNullOrEmpty(this.listenerConfiguration.ConsumerGroup) ? "$Default" : this.listenerConfiguration.ConsumerGroup;
+            this.topicName = this.listenerConfiguration.Topic;
+            this.functionId = functionDescriptorId;
+            Init();
         }
 
-        public void Cancel()
-        {
-            this.SafeCloseConsumerAsync().GetAwaiter().GetResult();
-        }
-
-        public Task StartAsync(CancellationToken cancellationToken)
+        private void Init()
```

Comment:
Looks like you chose to do all this init on construction rather than when the listener is started? Why? I don't know all the init those calls do, but it looks like it's starting up the Kafka client, subscribing, etc. which might not be construction time operations?

Reply:
Done